tensorflow-frameworks/tensorflow-data/src/test/java/org/tensorflow/data/DatasetOpTester.java

```diff
@@ -0,0 +1,283 @@
+package org.tensorflow.data;
+
+import org.junit.Test;
+import org.tensorflow.*;
+import org.tensorflow.op.Ops;
+import org.tensorflow.op.core.Constant;
+import org.tensorflow.op.data.*;
+import org.tensorflow.tools.Shape;
+import org.tensorflow.types.TInt32;
+import org.tensorflow.utils.Tuple2;
+
+import java.nio.IntBuffer;
+import java.util.Arrays;
+import java.util.List;
+
+public class DatasetOpTester {
+
+  @Test
+  public void testEagerBatching() {
+    try (EagerSession session = EagerSession.create()) {
+      Ops tf = Ops.create(session);
+
+      Constant<TInt32> X = tf.val(
+          new int[][]{
+              {1, 2, 3},
+              {4, 5, 6},
+              {7, 8, 9},
+              {10, 11, 12}
+          }
+      );
+
+      Constant<TInt32> y = tf.val(
+          new int[][]{
+              {1},
+              {4},
+              {7},
+              {10}
+          }
+      );
+
+
+      List<Operand<?>> tensors = Arrays.asList(X, y);
+
+      // // Try running TensorDataset
+      List<DataType<?>> outputTypes = Arrays.asList(TInt32.DTYPE, TInt32.DTYPE);
+      List<Shape> outputShapes = Arrays.asList(
+          Shape.of(3),
+          Shape.of(1));
+
+      TensorSliceDataset tensorDataset = TensorSliceDataset.create(tf.scope(), tensors, outputShapes);
+
+      BatchDataset batchDataset = BatchDataset.create(
```

Comment:
Are `TensorSliceDataset`, `BatchDataset`, etc. meant to be accessed directly or this is only for testing purposes? 

If they are, then you'll probably want to move these out of the `impl` package

Reply:
They shouldn't be accessed directly. This file was just for testing, so I removed it.